packages/gatsby/src/redux/actions/public.ts

```diff
@@ -1112,21 +1143,25 @@ actions.replaceWebpackConfig = (config: Object, plugin?: ?Plugin = null) => {
  *   }
  * })
  */
-actions.setBabelOptions = (options: Object, plugin?: ?Plugin = null) => {
+actions.setBabelOptions = (
+  options: Record<string, any>,
+  plugin: IGatsbyPlugin | null = null
```

Comment:
I looked through the codebase, this is not actually ever not defined. And it looks like our previous code didn't even handle a null properly. Let's just make this tight `plugin: IGatsbyPlugin`

Reply:
Done in 89f459b